silhouette/test/com/mohiva/play/silhouette/impl/providers/oauth2/Auth0ProviderSpec.scala

```diff
@@ -0,0 +1,175 @@
+/**
+ * Copyright 2015 Mohiva Organisation (license at mohiva dot com)
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.mohiva.play.silhouette.impl.providers.oauth2
+
+import com.mohiva.play.silhouette.api.LoginInfo
+import com.mohiva.play.silhouette.impl.exceptions.{ ProfileRetrievalException, UnexpectedResponseException }
+import com.mohiva.play.silhouette.impl.providers.OAuth2Provider._
+import com.mohiva.play.silhouette.impl.providers._
+import play.api.libs.json.{ JsArray, JsValue, Json }
+import play.api.libs.ws.{ WSRequest, WSResponse }
+import play.api.mvc.Results
+import play.api.test.{ FakeRequest, WithApplication }
+import play.mvc.Http
+import test.Helper
+
+import scala.concurrent.Future
+
+/**
+ * Test case for the [[Auth0Provider]] class.
+ */
+class Auth0ProviderSpec extends OAuth2ProviderSpec {
+
+  val AUTH0_ERROR_JSON: String = "providers/custom/auth0.error.json"
```

Comment:
I know, the types declarations are introduced by IntelliJ when you extract a constant ... hate it so much :-(
Done.